broker/src/main/java/org/apache/rocketmq/broker/failover/EscapeBridge.java

```diff
@@ -264,6 +262,44 @@ public PutMessageResult putMessageToSpecificQueue(MessageExtBrokerInner messageE
         }
     }
 
+    public CompletableFuture<PutMessageResult> asyncPutMessageToSpecificQueue(MessageExtBrokerInner messageExt) {
+        BrokerController masterBroker = this.brokerController.peekMasterBroker();
+        if (masterBroker != null) {
```

Comment:
There is a lot of code duplication with putMessageToSpecificQueue, which can be refactored so that both methods share a single implementation